Helidon 4.x cannot be built on Windows

After merging Pico, it is not possible to build Helidon on Windows.

```
dub 27, 2023 4:05:54 ODP. io.helidon.pico.tools.AbstractFilerMessager error
SEVERE: Failed to write resource file: Invalid relative name: META-INF\services\io.helidon.pico.api.ModuleComponent
io.helidon.pico.tools.ToolsException: Failed to write resource file: Invalid relative name: META-INF\services\io.helidon.pico.api.ModuleComponent
        at io.helidon.pico.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:286)
        at io.helidon.pico.tools.CodeGenFiler.codegenMetaInfServices(CodeGenFiler.java:194)
        at io.helidon.pico.tools.DefaultActivatorCreator.codegenMetaInfServices(DefaultActivatorCreator.java:300)
        at io.helidon.pico.tools.DefaultActivatorCreator.codegen(DefaultActivatorCreator.java:202)
        at io.helidon.pico.tools.DefaultActivatorCreator.createModuleActivators(DefaultActivatorCreator.java:124)
        at io.helidon.pico.processor.BaseAnnotationProcessor.doFiler(BaseAnnotationProcessor.java:632)
        at io.helidon.pico.processor.BaseAnnotationProcessor.process(BaseAnnotationProcessor.java:189)
        at io.helidon.pico.processor.ServiceAnnotationProcessor.process(ServiceAnnotationProcessor.java:108)
        at io.helidon.pico.configdriven.processor.ConfiguredByProcessor.process(ConfiguredByProcessor.java:92)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1018)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs(JavacProcessingEnvironment.java:854)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1260)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1399)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1248)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:930)
        at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:317)
        at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:176)
        at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:82)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess0(JavacCompiler.java:694)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcessWithProperClassloader(JavacCompiler.java:675)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess(JavacCompiler.java:664)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:186)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1209)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:198)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:342)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:330)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:175)
        at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:76)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:163)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:160)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:827)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:272)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:195)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: java.lang.IllegalArgumentException: Invalid relative name: META-INF\services\io.helidon.pico.api.ModuleComponent
        at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getFileForOutput(JavacFileManager.java:894)
        at java.compiler@20/javax.tools.JavaFileManager.getFileForOutputForOriginatingFiles(JavaFileManager.java:533)
        at jdk.compiler/com.sun.tools.javac.main.DelegatingJavaFileManager.getFileForOutputForOriginatingFiles(DelegatingJavaFileManager.java:139)
        at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createResource(JavacFiler.java:551)
        at io.helidon.pico.tools.AbstractFilerMessager.createResource(AbstractFilerMessager.java:121)
        at io.helidon.pico.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:256)
        ... 50 more

dub 27, 2023 4:05:54 ODP. io.helidon.pico.processor.BaseAnnotationProcessor error
SEVERE: ConfiguredByProcessor: Error: Error detected while processing [io.helidon.nima.webserver.LoomServer]
io.helidon.pico.tools.ToolsException: Error detected while processing [io.helidon.nima.webserver.LoomServer]
        at io.helidon.pico.processor.BaseAnnotationProcessor.doFiler(BaseAnnotationProcessor.java:645)
        at io.helidon.pico.processor.BaseAnnotationProcessor.process(BaseAnnotationProcessor.java:189)
        at io.helidon.pico.processor.ServiceAnnotationProcessor.process(ServiceAnnotationProcessor.java:108)
        at io.helidon.pico.configdriven.processor.ConfiguredByProcessor.process(ConfiguredByProcessor.java:92)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1018)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs(JavacProcessingEnvironment.java:854)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1260)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1399)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1248)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:930)
        at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:317)
        at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:176)
        at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:82)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess0(JavacCompiler.java:694)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcessWithProperClassloader(JavacCompiler.java:675)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.compileInProcess(JavacCompiler.java:664)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:186)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1209)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:198)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:342)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:330)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:175)
        at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:76)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:163)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:160)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:827)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:272)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:195)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: io.helidon.pico.tools.ToolsException: error during codegen
        at io.helidon.pico.processor.BaseAnnotationProcessor.doFiler(BaseAnnotationProcessor.java:634)
        ... 45 more
Caused by: io.helidon.pico.tools.ToolsException: Failed in create
        at io.helidon.pico.tools.DefaultActivatorCreator.createModuleActivators(DefaultActivatorCreator.java:130)
        at io.helidon.pico.processor.BaseAnnotationProcessor.doFiler(BaseAnnotationProcessor.java:632)
        ... 45 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.toPath()" because the return value of "java.util.concurrent.atomic.AtomicReference.get()" is null
        at io.helidon.pico.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:290)
        at io.helidon.pico.tools.CodeGenFiler.codegenMetaInfServices(CodeGenFiler.java:194)
        at io.helidon.pico.tools.DefaultActivatorCreator.codegenMetaInfServices(DefaultActivatorCreator.java:300)
        at io.helidon.pico.tools.DefaultActivatorCreator.codegen(DefaultActivatorCreator.java:202)
        at io.helidon.pico.tools.DefaultActivatorCreator.createModuleActivators(DefaultActivatorCreator.java:124)
        ... 46 more

```

```
io.helidon.inject.tools.ToolsException: Error writing resource file: java.lang.IllegalArgumentException: Invalid relative name: META-INF\services\io.helidon.inject.api.ModuleComponent
	at io.helidon.inject.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:521)
	at io.helidon.inject.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:240)
	at io.helidon.inject.tools.CodeGenFiler.codegenMetaInfServices(CodeGenFiler.java:227)
	at io.helidon.inject.tools.ActivatorCreatorDefault.codegenMetaInfServices(ActivatorCreatorDefault.java:427)
	at io.helidon.inject.tools.ActivatorCreatorDefault.codegen(ActivatorCreatorDefault.java:411)
	at io.helidon.inject.tools.ActivatorCreatorDefault.createModuleActivators(ActivatorCreatorDefault.java:330)
	at io.helidon.inject.processor.CreatorHandler.createModuleActivators(CreatorHandler.java:54)
	at io.helidon.inject.processor.InjectionAnnotationProcessor.doFiler(InjectionAnnotationProcessor.java:296)
	at io.helidon.inject.processor.InjectionAnnotationProcessor.doProcess(InjectionAnnotationProcessor.java:205)
	at io.helidon.inject.processor.InjectionAnnotationProcessor.process(InjectionAnnotationProcessor.java:173)
	at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
	at org.gradle.api.internal.tasks.compile.processing.NonIncrementalProcessor.process(NonIncrementalProcessor.java:45)
	at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.access$401(TimeTrackingProcessor.java:37)
	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:99)
	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:96)
	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.track(TimeTrackingProcessor.java:117)
	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.process(TimeTrackingProcessor.java:96)
	at app/jdk.compiler@21/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1021)
	at app/jdk.compiler@21/com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs(JavacProcessingEnvironment.java:857)
	at app/jdk.compiler@21/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1263)
	at app/jdk.compiler@21/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1402)
	at app/jdk.compiler@21/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1271)
	at app/jdk.compiler@21/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:948)
	at app/jdk.compiler@21/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
	at app/jdk.compiler@21/com.sun.tools.javac.api.JavacTaskImpl.invocationHelper(JavacTaskImpl.java:152)
	at app/jdk.compiler@21/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
	at app/jdk.compiler@21/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
	at org.gradle.internal.compiler.java.IncrementalCompileTask.call(IncrementalCompileTask.java:92)
	at org.gradle.api.internal.tasks.compile.AnnotationProcessingCompileTask.call(AnnotationProcessingCompileTask.java:94)
	at org.gradle.api.internal.tasks.compile.ResourceCleaningCompilationTask.call(ResourceCleaningCompilationTask.java:57)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:55)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:39)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:98)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:52)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:38)
	at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:51)
	at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:37)
	at org.gradle.api.internal.tasks.compile.ModuleApplicationNameWritingCompiler.execute(ModuleApplicationNameWritingCompiler.java:46)
	at org.gradle.api.internal.tasks.compile.ModuleApplicationNameWritingCompiler.execute(ModuleApplicationNameWritingCompiler.java:36)
	at org.gradle.jvm.toolchain.internal.DefaultToolchainJavaCompiler.execute(DefaultToolchainJavaCompiler.java:57)
	at org.gradle.api.tasks.compile.JavaCompile.lambda$createToolchainCompiler$3(JavaCompile.java:202)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompiler.execute(CleaningJavaCompiler.java:53)
	at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilerFactory.lambda$createRebuildAllCompiler$0(IncrementalCompilerFactory.java:52)
	at org.gradle.api.internal.tasks.compile.incremental.SelectiveCompiler.execute(SelectiveCompiler.java:70)
	at org.gradle.api.internal.tasks.compile.incremental.SelectiveCompiler.execute(SelectiveCompiler.java:44)
	at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:66)
	at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:52)
	at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler$1.call(CompileJavaBuildOperationReportingCompiler.java:64)
	at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler$1.call(CompileJavaBuildOperationReportingCompiler.java:48)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler.execute(CompileJavaBuildOperationReportingCompiler.java:48)
	at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:220)
	at org.gradle.api.tasks.compile.JavaCompile.performIncrementalCompilation(JavaCompile.java:161)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:146)
	at java.base@21/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base@21/java.lang.reflect.Method.invoke(Method.java:580)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:45)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.execute(IncrementalTaskAction.java:26)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:248)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:233)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:216)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:199)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:166)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:56)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:67)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:37)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:50)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:28)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.executeDelegateBroadcastingChanges(CaptureStateAfterExecutionStep.java:100)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:72)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:50)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:40)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:29)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:179)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:70)
	at org.gradle.internal.Either$Right.fold(Either.java:175)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:68)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:46)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:25)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:91)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:55)
	at java.base@21/java.util.Optional.orElseGet(Optional.java:364)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:37)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:65)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:36)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:77)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:38)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:94)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:49)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:71)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:45)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.executeWithNonEmptySources(SkipEmptyWorkStep.java:177)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:86)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:53)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:32)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:21)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
	at org.gradle.internal.execution.steps.CleanupStaleOutputsStep.execute(CleanupStaleOutputsStep.java:75)
	at org.gradle.internal.execution.steps.CleanupStaleOutputsStep.execute(CleanupStaleOutputsStep.java:41)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:66)
	at java.base@21/java.util.Optional.orElseGet(Optional.java:364)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:66)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:38)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:32)
	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:293)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:21)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:47)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:34)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:64)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:146)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:135)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.base@21/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base@21/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base@21/java.lang.Thread.run(Thread.java:1583)
Caused by: com.sun.tools.javac.util.ClientCodeException: java.lang.IllegalArgumentException: Invalid relative name: META-INF\services\io.helidon.inject.api.ModuleComponent
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedJavaFileManager.getFileForOutputForOriginatingFiles(ClientCodeWrapper.java:337)
	at java.compiler@21/javax.tools.ForwardingJavaFileManager.getFileForOutputForOriginatingFiles(ForwardingJavaFileManager.java:228)
	at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.getResource(JavacFiler.java:592)
	at io.helidon.inject.tools.AbstractFilerMessager.getResource(AbstractFilerMessager.java:134)
	at io.helidon.inject.tools.CodeGenFiler.codegenResourceFilerOut(CodeGenFiler.java:496)
	... 188 more
Caused by: java.lang.IllegalArgumentException: Invalid relative name: META-INF\services\io.helidon.inject.api.ModuleComponent
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getFileForOutput(JavacFileManager.java:896)
	at java.compiler@21/javax.tools.JavaFileManager.getFileForOutputForOriginatingFiles(JavaFileManager.java:535)
	at java.compiler@21/javax.tools.ForwardingJavaFileManager.getFileForOutputForOriginatingFiles(ForwardingJavaFileManager.java:228)
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedJavaFileManager.getFileForOutputForOriginatingFiles(ClientCodeWrapper.java:333)
	... 192 more
```

I also encountered an error when using helidon inject on Windows that cannot be compiled. I am not sure if this is the same issue.

--- a/.github/actions/common/action.yml
+++ b/.github/actions/common/action.yml
@@ -54,7 +54,7 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - if: ${{ inputs.free-space == 'true' }}
+    - if: ${{ inputs.free-space == 'true' && runner.os != 'Windows'}}
       # See https://github.com/actions/runner-images/issues/2840
       name: Free disk space
       shell: bash
@@ -141,7 +141,7 @@ runs:
       shell: bash
     - name: Archive test results
       # https://github.com/actions/upload-artifact/issues/240
-      if: ${{ inputs.test-artifact-name != '' && runner.os != 'Windows' && always() }}
+      if: ${{ inputs.test-artifact-name != '' && always() }}
       uses: actions/upload-artifact@v4
       with:
         if-no-files-found: 'ignore'

--- a/.github/actions/common/action.yml
+++ b/.github/actions/common/action.yml
@@ -140,7 +140,6 @@ runs:
       run: ${{ inputs.run }}
       shell: bash
     - name: Archive test results
-      # https://github.com/actions/upload-artifact/issues/240
       if: ${{ inputs.test-artifact-name != '' && always() }}
       uses: actions/upload-artifact@v4
       with:

--- a/.github/workflows/validate.yml
+++ b/.github/workflows/validate.yml
@@ -99,12 +99,13 @@ jobs:
     timeout-minutes: 30
     strategy:
       matrix:
-        os: [ ubuntu-20.04 ]
+        os: [ ubuntu-20.04, windows-2022 ]
         moduleSet: [ core, it, dbclient, dbclient-oracle, others ]
         include:
           - { os: ubuntu-20.04, platform: linux }
+          - { os: windows-2022, platform: windows }
     runs-on: ${{ matrix.os }}
-    name: tests/${{ matrix.moduleSet }}
+    name: tests/${{ matrix.moduleSet }}-${{matrix.platform}}
     steps:
       - uses: actions/checkout@v4
         with:
@@ -112,7 +113,7 @@ jobs:
       - uses: ./.github/actions/common
         with:
           build-cache: read-only
-          test-artifact-name: tests-${{ matrix.moduleSet }}
+          test-artifact-name: tests-${{ matrix.moduleSet }}-${{ matrix.platform }}
           run: |
             mvn ${MAVEN_ARGS} \
               -DreactorRule=tests \
@@ -311,11 +312,12 @@ jobs:
     timeout-minutes: 30
     strategy:
       matrix:
-        os: [ ubuntu-20.04, macos-14 ]
+        os: [ ubuntu-20.04, macos-14, windows-2022 ]
         packaging: [ jar, jlink ]
         include:
           - { os: ubuntu-20.04, platform: linux }
           - { os: macos-14, platform: macos }
+          - { os: windows-2022, platform: windows }
     runs-on: ${{ matrix.os }}
     name: tests/packaging-${{ matrix.packaging }}-${{ matrix.platform }}
     steps:
@@ -337,11 +339,12 @@ jobs:
     timeout-minutes: 30
     strategy:
       matrix:
-        os: [ ubuntu-20.04, macos-14 ]
+        os: [ ubuntu-20.04, macos-14, windows-2022 ]
         module: [ mp-1, mp-2, mp-3, se-1, inject ]
         include:
           - { os: ubuntu-20.04, platform: linux }
           - { os: macos-14, platform: macos }
+          - { os: windows-2022, platform: windows }
     runs-on: ${{ matrix.os }}
     name: tests/native-image-${{ matrix.module }}-${{ matrix.platform }}
     steps:

--- a/builder/tests/builder/src/test/java/io/helidon/builder/test/CustomNamedTest.java
+++ b/builder/tests/builder/src/test/java/io/helidon/builder/test/CustomNamedTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2022, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -56,7 +56,11 @@ class CustomNamedTest {
                 .build();
         DefaultPrettyPrinter printer = new DefaultPrettyPrinter();
         String json = mapper.writer(printer).writeValueAsString(customNamed);
-        assertThat(json, equalTo("{\n" + "  \"stringSet\" : [ \"b\", \"a\", \"y\" ]\n" + "}"));
+        assertThat(json, equalTo("{"
+                + System.lineSeparator()
+                + "  \"stringSet\" : [ \"b\", \"a\", \"y\" ]"
+                + System.lineSeparator()
+                + "}"));
     }
 
 }

--- a/integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciAuthenticationDetailsProvider.java
+++ b/integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciAuthenticationDetailsProvider.java
@@ -16,6 +16,7 @@
 
 package io.helidon.integrations.oci.sdk.runtime;
 
+import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Path;
@@ -159,7 +160,14 @@ class OciAuthenticationDetailsProvider implements Service.InjectionPointFactory<
     }
 
     static String userHomePrivateKeyPath(OciConfig ociConfig) {
-        return Paths.get(System.getProperty("user.home"), ".oci", ociConfig.authKeyFile()).toString();
+        return normalizePath(Paths.get(System.getProperty("user.home"), ".oci", ociConfig.authKeyFile()).toString());
+    }
+
+    private static String normalizePath(String value) {
+        if (value == null) {
+            return null;
+        }
+        return value.replace(File.separator, "/");
     }
 
 

--- a/microprofile/telemetry/src/test/java/io/helidon/microprofile/telemetry/TestTracerAtStartup.java
+++ b/microprofile/telemetry/src/test/java/io/helidon/microprofile/telemetry/TestTracerAtStartup.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -19,6 +19,8 @@ import io.helidon.microprofile.testing.junit5.AddConfig;
 import io.helidon.microprofile.testing.junit5.HelidonTest;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
@@ -29,6 +31,7 @@ import static org.hamcrest.Matchers.not;
 class TestTracerAtStartup {
 
     @Test
+    @DisabledOnOs(value = OS.WINDOWS, disabledReason = "https://github.com/helidon-io/helidon/issues/9513")
     void checkForFullFeaturedTracerAtStartup() {
         assertThat("Global tracer from start-up extension",
                    TestExtension.globalTracerAtStartup.unwrap(io.opentelemetry.api.trace.Tracer.class).getClass().getName(),

--- a/microprofile/testing/testng/src/main/java/io/helidon/microprofile/testing/testng/HelidonTestNgListener.java
+++ b/microprofile/testing/testng/src/main/java/io/helidon/microprofile/testing/testng/HelidonTestNgListener.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2022, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -106,6 +106,7 @@ public class HelidonTestNgListener implements IClassListener, ITestListener {
         testClass = iTestClass.getRealClass();
         HelidonTest testAnnot = testClass.getAnnotation(HelidonTest.class);
         if (testAnnot == null) {
+            helidonTest = false;
             return;
         }
         helidonTest = true;

--- a/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/InMemorySpanExporter.java
+++ b/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/InMemorySpanExporter.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,25 +13,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.helidon.microprofile.telemetry;
-
-import static java.util.Comparator.comparingLong;
-import static java.util.concurrent.TimeUnit.SECONDS;
-import static org.hamcrest.MatcherAssert.assertThat;
 
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.stream.Collectors;
 
-import org.awaitility.Awaitility;
-
 import io.opentelemetry.sdk.common.CompletableResultCode;
 import io.opentelemetry.sdk.trace.data.SpanData;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import jakarta.enterprise.context.ApplicationScoped;
+import org.awaitility.Awaitility;
 import org.hamcrest.Matchers;
+
+import static java.util.Comparator.comparingLong;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 @ApplicationScoped
 public class InMemorySpanExporter implements SpanExporter {
@@ -48,7 +46,8 @@ public class InMemorySpanExporter implements SpanExporter {
     }
 
     public void assertSpanCount(int spanCount) {
-        Awaitility.await().pollDelay(3, SECONDS).atMost(10, SECONDS)
+        Awaitility.await()
+                .pollDelay(3, SECONDS).atMost(10, SECONDS)
                 .untilAsserted(() -> assertThat(finishedSpanItems.size(), Matchers.is(spanCount)));
     }
 
@@ -57,6 +56,7 @@ public class InMemorySpanExporter implements SpanExporter {
     }
 
     @Override
+    @SuppressWarnings("NullableProblems")
     public CompletableResultCode export(Collection<SpanData> spans) {
         if (isStopped) {
             return CompletableResultCode.ofFailure();

--- a/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/InMemorySpanExporterProvider.java
+++ b/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/InMemorySpanExporterProvider.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.helidon.microprofile.telemetry;
 
 import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
@@ -21,6 +20,7 @@ import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterPro
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import jakarta.enterprise.inject.spi.CDI;
 
+@SuppressWarnings("NullableProblems")
 public class InMemorySpanExporterProvider implements ConfigurableSpanExporterProvider {
     @Override
     public SpanExporter createExporter(final ConfigProperties config) {

--- a/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/RestSpanHierarchyTest.java
+++ b/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/RestSpanHierarchyTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,31 +13,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-
 package io.helidon.microprofile.telemetry;
 
 import java.util.List;
 
-import io.helidon.http.Status;
-import io.helidon.microprofile.telemetry.InMemorySpanExporterProvider;
 import io.helidon.microprofile.testing.junit5.AddBean;
-import io.helidon.microprofile.testing.junit5.AddConfig;
-import io.helidon.microprofile.testing.junit5.AddExtension;
+import io.helidon.microprofile.testing.junit5.AddConfigBlock;
 import io.helidon.microprofile.testing.junit5.HelidonTest;
+import io.helidon.tracing.Span;
+import io.helidon.tracing.Tracer;
 
 import io.opentelemetry.api.common.AttributeKey;
 import io.opentelemetry.instrumentation.annotations.WithSpan;
 import io.opentelemetry.sdk.trace.data.SpanData;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
-import jakarta.ws.rs.ApplicationPath;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.client.WebTarget;
-import jakarta.ws.rs.core.Application;
 import jakarta.ws.rs.core.Response;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
@@ -53,10 +47,12 @@ import static org.hamcrest.Matchers.is;
 @AddBean(RestSpanHierarchyTest.SpanResource.class)
 @AddBean(InMemorySpanExporter.class)
 @AddBean(InMemorySpanExporterProvider.class)
-@AddConfig(key = "otel.service.name", value = "helidon-mp-telemetry")
-@AddConfig(key = "otel.sdk.disabled", value = "false")
-@AddConfig(key = "telemetry.span.full.url", value = "false")
-@AddConfig(key = "otel.traces.exporter", value = "in-memory")
+@AddConfigBlock("""
+        otel.service.name=helidon-mp-telemetry
+        otel.sdk.disabled=false
+        otel.traces.exporter=in-memory
+        telemetry.span.full.url=false
+        """)
 public class RestSpanHierarchyTest {
 
     @Inject
@@ -72,7 +68,6 @@ public class RestSpanHierarchyTest {
 
     @Test
     void spanHierarchy() {
-
         assertThat(webTarget.path("mixed").request().get().getStatus(), is(Response.Status.OK.getStatusCode()));
 
         List<SpanData> spanItems = spanExporter.getFinishedSpanItems(4);
@@ -81,11 +76,9 @@ public class RestSpanHierarchyTest {
         assertThat(spanItems.get(0).getAttributes().get(AttributeKey.stringKey("attribute")), is("value"));
         assertThat(spanItems.get(0).getParentSpanId(), is(spanItems.get(1).getSpanId()));
 
-
         assertThat(spanItems.get(1).getKind(), is(INTERNAL));
         assertThat(spanItems.get(1).getName(), is("mixed_parent"));
         assertThat(spanItems.get(1).getParentSpanId(), is(spanItems.get(2).getSpanId()));
-
 
         assertThat(spanItems.get(2).getKind(), is(SERVER));
         assertThat(spanItems.get(2).getName(), is("/mixed"));
@@ -119,17 +112,16 @@ public class RestSpanHierarchyTest {
     public static class SpanResource {
 
         @Inject
-        private io.helidon.tracing.Tracer helidonTracerInjected;
-
+        private Tracer helidonTracerInjected;
 
         @GET
         @Path("mixed")
         @WithSpan("mixed_parent")
         public Response mixedSpan() {
 
-            io.helidon.tracing.Tracer helidonTracer = io.helidon.tracing.Tracer.global();
-            io.helidon.tracing.Span mixedSpan = helidonTracer.spanBuilder("mixed_inner")
-                    .kind(io.helidon.tracing.Span.Kind.SERVER)
+            Tracer helidonTracer = Tracer.global();
+            Span mixedSpan = helidonTracer.spanBuilder("mixed_inner")
+                    .kind(Span.Kind.SERVER)
                     .tag("attribute", "value")
                     .start();
             mixedSpan.end();
@@ -142,8 +134,8 @@ public class RestSpanHierarchyTest {
         @WithSpan("mixed_parent_injected")
         public Response mixedSpanInjected() {
 
-            io.helidon.tracing.Span mixedSpan = helidonTracerInjected.spanBuilder("mixed_inner_injected")
-                    .kind(io.helidon.tracing.Span.Kind.SERVER)
+            Span mixedSpan = helidonTracerInjected.spanBuilder("mixed_inner_injected")
+                    .kind(Span.Kind.SERVER)
                     .tag("attribute", "value")
                     .start();
             mixedSpan.end();

--- a/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/TestFullUrlName.java
+++ b/microprofile/tests/telemetry/src/test/java/io/helidon/microprofile/telemetry/TestFullUrlName.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,32 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.helidon.microprofile.telemetry;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
-import io.helidon.http.Status;
-import io.helidon.microprofile.telemetry.InMemorySpanExporter;
-import io.helidon.microprofile.telemetry.InMemorySpanExporterProvider;
 import io.helidon.microprofile.testing.junit5.AddBean;
 import io.helidon.microprofile.testing.junit5.AddConfig;
+import io.helidon.microprofile.testing.junit5.AddConfigBlock;
 import io.helidon.microprofile.testing.junit5.AddExtension;
 import io.helidon.microprofile.testing.junit5.HelidonTest;
 
 import io.opentelemetry.sdk.trace.data.SpanData;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
-import jakarta.ws.rs.ApplicationPath;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.client.WebTarget;
-import jakarta.ws.rs.core.Application;
 import jakarta.ws.rs.core.Response;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
 import static org.hamcrest.Matchers.is;
 
 /**
@@ -49,16 +46,16 @@ import static org.hamcrest.Matchers.is;
 @AddBean(InMemorySpanExporter.class)
 @AddBean(InMemorySpanExporterProvider.class)
 @AddExtension(TelemetryCdiExtension.class)
-@AddConfig(key = "otel.service.name", value = "helidon-mp-telemetry")
-@AddConfig(key = "otel.sdk.disabled", value = "false")
-@AddConfig(key = "telemetry.span.full.url", value = "true")
-@AddConfig(key = "otel.traces.exporter", value = "in-memory")
+@AddConfigBlock("""
+        otel.service.name=helidon-mp-telemetry
+        otel.sdk.disabled=false
+        otel.traces.exporter=in-memory
+        telemetry.span.full.url=true
+        """)
 public class TestFullUrlName {
-
 
     @Inject
     WebTarget webTarget;
-
 
     @Inject
     InMemorySpanExporter spanExporter;
@@ -73,12 +70,16 @@ public class TestFullUrlName {
 
     @Test
     void spanNaming() {
-
         assertThat(webTarget.path("named").request().get().getStatus(), is(Response.Status.OK.getStatusCode()));
 
-        List<SpanData> spanItems = spanExporter.getFinishedSpanItems(2);
-        assertThat(spanItems.size(), is(2));
-        assertThat(spanItems.get(0).getName(), is("http://localhost:" + webTarget.getUri().getPort() + "/named"));
+        List<String> names = spanExporter.getFinishedSpanItems(2)
+                .stream()
+                .map(SpanData::getName)
+                .collect(Collectors.toList());
+
+        assertThat(names.size(), is(2));
+        assertThat(names, hasItem("http://localhost:" + webTarget.getUri().getPort() + "/named"));
+        assertThat(names, hasItem("HTTP GET"));
     }
 
 

--- a/tests/integration/harness/src/main/java/io/helidon/tests/integration/harness/ProcessRunner.java
+++ b/tests/integration/harness/src/main/java/io/helidon/tests/integration/harness/ProcessRunner.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -31,6 +31,7 @@ import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static java.util.Objects.requireNonNull;
 
@@ -467,7 +468,19 @@ public abstract class ProcessRunner {
         @Override
         protected List<String> command(List<String> opts, List<String> args) {
             Objects.requireNonNull(finalName, "finalName is null");
-            return new CommandBuilder("target/" + finalName + "-jri/bin/start" + (IS_WINDOWS ? ".ps1" : ""))
+            if (IS_WINDOWS) {
+                return new CommandBuilder("powershell")
+                        .append("-File")
+                        .append("target/" + finalName + "-jri/bin/start.ps1")
+                        .append("--jvm", opts.stream()
+                                .map(option -> String.format("'%s'", option))
+                                .collect(Collectors.joining(" ")))
+                        .append(args.stream()
+                                .map(argument -> String.format("'%s'", argument))
+                                .collect(Collectors.joining(" ")))
+                        .command();
+            }
+            return new CommandBuilder("target/" + finalName + "-jri/bin/start")
                     .append("--jvm", String.join(" ", opts))
                     .append(args)
                     .command();

--- a/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AckMpTest.java
+++ b/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AckMpTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -43,6 +43,8 @@ import org.junit.jupiter.api.MethodOrderer;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
 import static java.lang.System.Logger.Level.DEBUG;
 
@@ -106,6 +108,7 @@ public class AckMpTest extends AbstractMPTest {
 
     @Test
     @Order(2)
+    @DisabledOnOs(value = OS.WINDOWS, disabledReason = "https://github.com/helidon-io/helidon/issues/8509")
     void resendAckTestPart2(SeContainer cdi) {
             MockConnector mockConnector = cdi.select(MockConnector.class, TEST_CONNECTOR_ANNOTATION).get();
 

--- a/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/CommonLoginBase.java
+++ b/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/CommonLoginBase.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -54,7 +54,7 @@ import static org.hamcrest.CoreMatchers.startsWith;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.empty;
 
-@Testcontainers
+@Testcontainers(disabledWithoutDocker = true)
 @HelidonTest(resetPerTest = true)
 @AddBean(TestResource.class)
 class CommonLoginBase {

--- a/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/CookieBasedLoginIT.java
+++ b/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/CookieBasedLoginIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -20,8 +20,6 @@ import jakarta.ws.rs.client.Entity;
 import jakarta.ws.rs.client.WebTarget;
 import jakarta.ws.rs.core.Form;
 import jakarta.ws.rs.core.Response;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 import org.junit.jupiter.api.Test;
 
 import static io.helidon.tests.integration.oidc.TestResource.EXPECTED_POST_LOGOUT_TEST_MESSAGE;

--- a/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/IdTokenIT.java
+++ b/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/IdTokenIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -27,33 +27,24 @@ import io.helidon.microprofile.testing.junit5.AddConfig;
 import io.helidon.security.jwt.Jwt;
 import io.helidon.security.jwt.SignedJwt;
 import io.helidon.security.jwt.jwk.Jwk;
-import io.helidon.security.providers.oidc.common.OidcConfig;
 
 import jakarta.ws.rs.client.Client;
 import jakarta.ws.rs.client.ClientBuilder;
-import jakarta.ws.rs.client.Entity;
 import jakarta.ws.rs.client.Invocation;
 import jakarta.ws.rs.client.WebTarget;
-import jakarta.ws.rs.core.Form;
 import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.Response;
 import org.glassfish.jersey.client.ClientConfig;
 import org.glassfish.jersey.client.ClientProperties;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static io.helidon.security.providers.oidc.common.OidcConfig.DEFAULT_ID_COOKIE_NAME;
 import static io.helidon.tests.integration.oidc.TestResource.EXPECTED_TEST_MESSAGE;
-import static org.hamcrest.CoreMatchers.hasItem;
 import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.CoreMatchers.nullValue;
-import static org.hamcrest.CoreMatchers.startsWith;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.empty;
 
 class IdTokenIT extends CommonLoginBase {
 

--- a/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/QueryBasedLoginIT.java
+++ b/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/QueryBasedLoginIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -24,8 +24,6 @@ import jakarta.ws.rs.core.Form;
 import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.Response;
 import org.glassfish.jersey.client.ClientProperties;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 import org.junit.jupiter.api.Test;
 
 import static io.helidon.tests.integration.oidc.TestResource.EXPECTED_TEST_MESSAGE;

--- a/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/RefreshTokenIT.java
+++ b/tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/RefreshTokenIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -33,21 +33,15 @@ import jakarta.ws.rs.client.WebTarget;
 import jakarta.ws.rs.core.Form;
 import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.Response;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 import org.junit.jupiter.api.Test;
 
-import static io.helidon.tests.integration.oidc.TestResource.EXPECTED_POST_LOGOUT_TEST_MESSAGE;
 import static io.helidon.tests.integration.oidc.TestResource.EXPECTED_TEST_MESSAGE;
 import static org.hamcrest.CoreMatchers.hasItem;
-import static org.hamcrest.CoreMatchers.hasItems;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.CoreMatchers.not;
-import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.CoreMatchers.startsWith;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.empty;
 
 @AddConfig(key = "security.providers.1.oidc.token-signature-validation", value = "false")

--- a/tests/integration/packaging/mp-1/src/main/resources/META-INF/native-image/io.helidon.tests.integration.packaging/helidon-tests-integration-packaging-mp1/native-image.properties
+++ b/tests/integration/packaging/mp-1/src/main/resources/META-INF/native-image/io.helidon.tests.integration.packaging/helidon-tests-integration-packaging-mp1/native-image.properties
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+# Copyright (c) 2023, 2025 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -14,4 +14,5 @@
 # limitations under the License.
 #
 
-Args=--initialize-at-build-time=io.helidon.tests.integration.packaging.mp1
+Args=--initialize-at-build-time=io.helidon.tests.integration.packaging.mp1 \
+  --trace-object-instantiation=java.lang.Thread

--- a/tests/integration/packaging/mp-1/src/test/java/io/helidon/tests/integration/packaging/mp1/Mp1NativeTestIT.java
+++ b/tests/integration/packaging/mp-1/src/test/java/io/helidon/tests/integration/packaging/mp1/Mp1NativeTestIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -18,7 +18,12 @@ package io.helidon.tests.integration.packaging.mp1;
 import io.helidon.tests.integration.harness.ProcessRunner.ExecMode;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
+@DisabledOnOs(value = OS.WINDOWS,
+        disabledReason = "JFR event recording is not supported on windows yet " +
+                "https://www.graalvm.org/latest/reference-manual/native-image/debugging-and-diagnostics/JFR/")
 class Mp1NativeTestIT extends Mp1PackagingTestIT {
 
     @Override

--- a/tests/integration/packaging/mp-3/src/test/java/io/helidon/tests/integration/packaging/mp3/Mp3NativeTestIT.java
+++ b/tests/integration/packaging/mp-3/src/test/java/io/helidon/tests/integration/packaging/mp3/Mp3NativeTestIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -18,7 +18,12 @@ package io.helidon.tests.integration.packaging.mp3;
 import io.helidon.tests.integration.harness.ProcessRunner.ExecMode;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
+@DisabledOnOs(value = OS.WINDOWS,
+        disabledReason = "JFR event recording is not supported on windows yet " +
+                "https://www.graalvm.org/latest/reference-manual/native-image/debugging-and-diagnostics/JFR/")
 class Mp3NativeTestIT extends Mp3PackagingTestIT {
 
     @Override

--- a/webclient/tests/grpc/src/test/java/io/helidon/webclient/grpc/tests/GrpcClientUriTest.java
+++ b/webclient/tests/grpc/src/test/java/io/helidon/webclient/grpc/tests/GrpcClientUriTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -17,6 +17,7 @@
 package io.helidon.webclient.grpc.tests;
 
 import java.net.URI;
+import java.time.Duration;
 import java.util.concurrent.CountDownLatch;
 
 import io.helidon.common.configurable.Resource;
@@ -87,7 +88,10 @@ class GrpcClientUriTest extends GrpcBaseTest {
                 .build();
         StringServiceGrpc.StringServiceBlockingStub service = StringServiceGrpc.newBlockingStub(grpcClient.channel());
 
-        Retry retry = Retry.builder().calls(2).build();
+        Retry retry = Retry.builder()
+                .overallTimeout(Duration.ofMillis(5000))
+                .calls(2)
+                .build();
         Strings.StringMessage res = retry.invoke(() -> service.upper(newStringMessage("hello")));
         assertThat(res.getText(), is("HELLO"));
         assertThat(latch.getCount(), is(0L));

--- a/webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/FileSystemContentHandler.java
+++ b/webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/FileSystemContentHandler.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2017, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,6 +16,7 @@
 
 package io.helidon.webserver.staticcontent;
 
+import java.io.File;
 import java.io.IOException;
 import java.lang.System.Logger.Level;
 import java.nio.file.Files;
@@ -88,7 +89,7 @@ class FileSystemContentHandler extends FileBasedContentHandler {
 
         String rawPath = req.prologue().uriPath().rawPath();
 
-        String relativePath = root.relativize(path).toString().replace("\\", "/");
+        String relativePath = root.relativize(path).toString().replace(File.separator, "/");
         String requestedResource;
         if (mapped) {
             requestedResource = relativePath;

--- a/webserver/tests/http2/src/test/java/io/helidon/webserver/tests/http2/EmptyFrameCntTest.java
+++ b/webserver/tests/http2/src/test/java/io/helidon/webserver/tests/http2/EmptyFrameCntTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -60,6 +60,8 @@ import io.helidon.webserver.testing.junit5.SetUpRoute;
 import io.helidon.webserver.testing.junit5.SetUpServer;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
 import static io.helidon.http.Method.GET;
 import static io.netty.handler.codec.http2.Http2CodecUtil.FRAME_HEADER_LENGTH;
@@ -68,6 +70,7 @@ import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 
 @ServerTest
+@DisabledOnOs(value = OS.WINDOWS, disabledReason = "https://github.com/helidon-io/helidon/issues/8510")
 class EmptyFrameCntTest {
 
     private static final Duration TIMEOUT = Duration.ofSeconds(10);

--- a/webserver/tests/webserver/src/test/java/io/helidon/webserver/tests/WebServerStopIdleTest.java
+++ b/webserver/tests/webserver/src/test/java/io/helidon/webserver/tests/WebServerStopIdleTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2023, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -21,6 +21,9 @@ import io.helidon.webclient.http1.Http1Client;
 import io.helidon.webserver.WebServer;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
@@ -29,7 +32,20 @@ import static org.hamcrest.Matchers.lessThan;
 class WebServerStopIdleTest {
 
     @Test
+    @DisabledOnOs(OS.WINDOWS)
     void stopWhenIdleExpectTimelyStop() {
+        stopWhenIdleExpectTimelyStop(500);
+    }
+
+    @Test
+    @EnabledOnOs(
+            value = OS.WINDOWS,
+            disabledReason = "Slow pipeline runner make it stop a few millisecond later")
+    void stopWhenIdleExpectTimelyStopWindows() {
+        stopWhenIdleExpectTimelyStop(505);
+    }
+
+    void stopWhenIdleExpectTimelyStop(int timeout) {
         WebServer webServer = WebServer.builder()
                 .routing(router -> router.get("ok", (req, res) -> res.send("ok")))
                 .build();
@@ -46,6 +62,6 @@ class WebServerStopIdleTest {
         long startMillis = System.currentTimeMillis();
         webServer.stop();
         int stopExecutionTimeInMillis = (int) (System.currentTimeMillis() - startMillis);
-        assertThat(stopExecutionTimeInMillis, is(lessThan(500)));
+        assertThat(stopExecutionTimeInMillis, is(lessThan(timeout)));
     }
 }

--- a/webserver/tests/websocket/src/main/java/io/helidon/webserver/tests/websocket/WsConversationClient.java
+++ b/webserver/tests/websocket/src/main/java/io/helidon/webserver/tests/websocket/WsConversationClient.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2022, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -31,8 +31,6 @@ import static io.helidon.webserver.tests.websocket.WsAction.Operation.RCV;
 import static io.helidon.webserver.tests.websocket.WsAction.OperationType.BINARY;
 import static io.helidon.webserver.tests.websocket.WsAction.OperationType.TEXT;
 import static java.nio.charset.StandardCharsets.UTF_8;
-
-;
 
 /**
  * A websocket client that is driven by a conversation instance.

